### PR TITLE
Update results returned by `slice.items()`

### PR DESCRIPTION
**Pull request by jean-lucas**
Update the `GET /slice/:slice_id` endpoint to return `items` instead of `dataset_items`

The return type of `items` is based on the type of the slice: `scene, dataset_item, object`

### CODE REVIEW

**Review by gatli — CHANGES_REQUESTED** [review]
We have to keep the `items` property for backwards compatibility. Can you also make sure that `client.slices` populates the type field?

**Comment by gatli** [comment]
Feel free to skip the `_type` population refactor for now if you don't have time. Its just the `@property` that we have to adhere to that is blocking right now. I don't think anything is calling `_type` right now so listings should still be fast.

Also you can just merge after you've made the change. You have the rights! I just checked 🙂

File: nucleus/slice.py [hunk 1]
```diff
@@ -209,28 +219,70 @@ def items_generator(self, page_size=100000):
...
+                {}, f"slice/{self.id}", requests_command=requests.get
+            )
+        except NucleusAPIError as e:
+            if e.status_code == 503:
+                e.message += "/n Your request timed out while trying to get all the items in the slice. Please try slice.items_generator() instead."
+            raise e
+
+        dataset_item_jsons = response.get(
+            "dataset_items", []
+        )  # CHANGE ME BACK TO `ITEMS_KEY` BEFORE MERGE
```
[the remark this answers is not in the document]
Comment (jean-lucas) [reply nucleus/slice.py:241]: Need to merge the `scaleapi` backend first, and then I can change this back. 
Its just so the CI tests passes for now

File: nucleus/slice.py [hunk 2]
```diff
@@ -182,6 +185,13 @@ def dataset_id(self):
             self._dataset_id = self.info()["dataset_id"]
         return self._dataset_id
 
+    @property
+    def type(self):
+        """The type of the Slice."""
+        if self._type is None:
+            self._type = self.info()["type"]
```
[the remark this answers is not in the document]
Comment (gatli) [reply nucleus/slice.py:192]: The `client.slices` property hits `slice/` that returns all the relevant info that is then populated in the `Slice.from_request` classmethod. This avoids the `info/` roundtrip for large lists. If we create something that doesn't prepopulate the fields the call to `client.slices` can go from around 1-2s to 100s easily.

```
    @property
    def slices(self) -> List[Slice]:
        response = self.make_request({}, "slice/", requests.get)
        slices = [Slice.from_request(info, self) for info in response]
        return slices
```

**Review by gatli — APPROVED** [review]
LGTM 👍

**Pull request merged**